### PR TITLE
CB-16000 Fix flow engine to mark flow chain as failed if one of it's …

### PR DESCRIPTION
**Pull request by foldik**
…flow trigger condition fails.

See detailed description in the commit message.

### CODE REVIEW

**Review by lturcsanyi — APPROVED** [review]

File: flow/src/main/java/com/sequenceiq/cloudbreak/service/flowlog/FlowLogUtil.java [hunk 1]
```diff
@@ -67,6 +67,18 @@ public static boolean isFlowFailHandled(List<FlowLog> flowLogs, Set<String> fail
         return false;
     }
 
+    public static boolean isFlowInFailedState(List<FlowLog> flowLogs, Set<String> failHandledEvents) {
```
Comment (lacikaaa) [inline flow/src/main/java/com/sequenceiq/cloudbreak/service/flowlog/FlowLogUtil.java:70]: could you add unit tests?

File: flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java [hunk 2]
```diff
@@ -232,6 +236,21 @@ private AcceptResult handleNewFlowRequest(String key, Payload payload, FlowParam
         }
     }
 
+    private void createNewFailedFlow(String key, Payload payload, FlowParameters flowParameters, String flowChainId, FlowConfiguration<?> flowConfig)
```
Comment (lacikaaa) [inline flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java:239]: could you add some logs?

File: flow/src/main/java/com/sequenceiq/cloudbreak/service/flowlog/FlowLogUtil.java (hunk 1 above)
Comment (foldik) [reply flow/src/main/java/com/sequenceiq/cloudbreak/service/flowlog/FlowLogUtil.java:67]: sure

File: flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java (hunk 2 above)
Comment (foldik) [reply flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java:242]: yes

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**